test/src/nativeTest/kotlin/io/realm/InstrumentedTests.kt

```diff
@@ -39,30 +51,30 @@ class InstrumentedTests {
     @RealmModule(Sample::class)
     class MySchema
 
-    lateinit var realm: Realm
+    lateinit var tmpDir: String
 
     @BeforeTest
     fun setup() {
-        val configuration = RealmConfiguration.Builder(schema = MySchema()).build()
-        realm = Realm.open(configuration)
-        // FIXME Cleaning up realm to overcome lack of support for deleting actual files
-        //  https://github.com/realm/realm-kotlin/issues/95
-        realm.beginTransaction()
-        realm.objects(Sample::class).delete()
-        realm.commitTransaction()
-        assertEquals(0, realm.objects(Sample::class).size, "Realm is not empty")
+        tmpDir = createTempDir()
+    }
+
+    @AfterTest
+    fun tearDown() {
+        deleteTempDir(tmpDir)
     }
 
     // FIXME API-CLEANUP Do we actually want to expose this. Test should probably just be reeavluated
     //  or moved. Local implementation of pointer wrapper to support test. Using the internal one would
     //  require the native wrapper to be api dependency from cinterop/library. Don't know if the
     //  test is needed at all at this level
-    class CPointerWrapper(val ptr: CPointer<out CPointed>?) : NativePointer
+    class CPointerWrapper(val ptr: CPointer<out CPointed>?, managed: Boolean = true) : NativePointer
```

Comment:
This seems a bit dangerous, as the implementation now differs significantly. Given the FIXME it should maybe just go or be moved to an internal test in the compiler plugin...maybe something similar is already there.

Reply:
Let's discuss this in the refactor Sprint